registration/tests/models.py

```diff
@@ -471,17 +474,332 @@ def test_resend_activation_email_expired_user(self):
         Test the scenario where user tries to resend activation code
         to the expired user's email
         """
-        new_user = RegistrationProfile.objects.create_inactive_user(
+        new_user = self.registration_profile.objects.create_inactive_user(
             site=Site.objects.get_current(), send_email=False, **self.user_info)
         new_user.date_joined -= datetime.timedelta(
             days=settings.ACCOUNT_ACTIVATION_DAYS + 1)
         new_user.save()
 
-        profile = RegistrationProfile.objects.get(user=new_user)
+        profile = self.registration_profile.objects.get(user=new_user)
         self.assertTrue(profile.activation_key_expired())
 
-        self.assertFalse(RegistrationProfile.objects.resend_activation_mail(
+        self.assertFalse(self.registration_profile.objects.resend_activation_mail(
             email=self.user_info['email'],
             site=Site.objects.get_current(),
         ))
         self.assertEqual(len(mail.outbox), 0)
+
+
+class SupervisedRegistrationModelTests(RegistrationModelTests):
+    """
+    Test the model and manager used in the admin_approval backend.
+
+    """
+
+    user_info = {'username': 'alice',
+                 'password': 'swordfish',
+                 'email': 'alice@example.com'}
+
+    registration_profile = SupervisedRegistrationProfile
+
+    def setUp(self):
+        self.old_activation = getattr(settings,
+                                      'ACCOUNT_ACTIVATION_DAYS', None)
+        self.old_reg_email = getattr(settings,
+                                     'REGISTRATION_DEFAULT_FROM_EMAIL', None)
+        self.old_email_html = getattr(settings,
+                                      'REGISTRATION_EMAIL_HTML', None)
+        self.old_django_email = getattr(settings,
+                                        'DEFAULT_FROM_EMAIL', None)
+        self.old_admins = getattr(settings,
+                                  'ADMINS', None)
+
+        settings.ACCOUNT_ACTIVATION_DAYS = 7
+        settings.REGISTRATION_DEFAULT_FROM_EMAIL = 'registration@email.com'
+        settings.REGISTRATION_EMAIL_HTML = True
+        settings.DEFAULT_FROM_EMAIL = 'django@email.com'
+        settings.ADMINS = (
+            ('T-Rex', 'admin1@iamtrex.com'),
+            ('Flea', 'admin2@iamaflea.com')
+        )
+
+    def tearDown(self):
+        settings.ACCOUNT_ACTIVATION_DAYS = self.old_activation
+        settings.REGISTRATION_DEFAULT_FROM_EMAIL = self.old_reg_email
+        settings.REGISTRATION_EMAIL_HTML = self.old_email_html
+        settings.DEFAULT_FROM_EMAIL = self.old_django_email
+
+    def test_valid_activation(self):
```

Comment:
This is different than ``RegistrationModelTests.test_valid_activation()`` as the latter has the following:

    self.failUnless(activated.is_active)

When using the default backend, after activation ``UserModel.is_active`` is set to ``True`` but when using the admin_approval backend it is set to ``False`` until account is approved.